var/spack/repos/builtin/packages/su2/package.py

```diff
@@ -61,17 +61,56 @@ class Su2(MesonPackage):
     depends_on("openblas", when="+openblas ~mkl")
     depends_on("cmake", type="build", when="+mpp")
 
+    depends_on("codipack@:1.9.3", when="+autodiff")
+    depends_on("codipack@:1.9.3", when="+directdiff")
+    depends_on("medipack", when="+autodiff +mpi")
+    depends_on("medipack", when="+directdiff +mpi")
+    depends_on("opdilib", when="+autodiff +openmp")
+    depends_on("opdilib", when="+directdiff +openmp")
+    depends_on("codipack@openmp", when="+autodiff +openmp")
+    depends_on("codipack@openmp", when="+directdiff +openmp")
+
     # Remove the part that fixes the meson version to 0.61.1.
     # This fix is considered meaningless and will be removed in the next version(@7.6:) of SU2.
     patch("meson_version.patch", when="@7.4.0:7.5.1")
 
+    def patch(self):
+        if "+autodiff" in self.spec or "+directdiff" in self.spec:
+            filter_file(
+                "externals/codi/include",
+                join_path(self.spec["codipack"].prefix, "include"),
+                "meson.build",
+            )
+
+        if ("+autodiff" in self.spec or "+directdiff" in self.spec) and "+mpi" in self.spec:
+            filter_file(
+                "externals/medi/include",
+                join_path(self.spec["medipack"].prefix, "include"),
```

Comment:
FYI.  Since Spack's `prefix` is special (see docs at https://spack.readthedocs.io/en/latest/_modules/spack/util/prefix.html#), you could just use:

```suggestion
                self.spec["medipack"].prefix.include,
```